File: packages/core/test-fixtures/DynamicNames.java
import software.amazon.lambda.durable.DurableContext;
import software.amazon.lambda.durable.DurableHandler;

public class DynamicNames extends DurableHandler<String, String> {
    protected String handleRequest(String input, DurableContext ctx) {
        ctx.step("string-literal", String.class, c -> doWork());
        String compName = "runtime";
        ctx.step(compName, String.class, c -> undo());
        ctx.step(getCompName(), String.class, c -> undo());
        ctx.step(
            "multi-line-name", String.class,
            c -> doStuff()
        );
        ctx.step(comp.name, String.class, c -> undo());
        return "";
    }
}

File: packages/core/test-fixtures/dynamic-names.py
from aws_durable_execution_sdk_python import DurableContext, durable_execution

def do_work(x): return x

@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    context.step(do_work(event["id"]))
    context.step(do_work(event["id"]), name="explicit-name")
    comp_name = "runtime-name"
    context.step(do_work(event["id"]), name=comp_name)
